pkg/bmt/proof.go

```diff
@@ -0,0 +1,64 @@
+package bmt
+
+// Prover wraps the Hasher to allow Merkle proof functionality
+type Prover struct {
+	*Hasher
+}
+
+// Proof represents a Merkle proof of segment
+type Proof struct {
+	Section []byte
+	Sisters [][]byte
+	Span    []byte
+}
+
+// Proof returns the inclusion proof of the i-th data segment
+func (p Prover) Proof(i int) Proof {
+	i = i / 2
```

Comment:
I would add some input validation here to check that we never have more than 128 segments here, otherwise we will panic